jax/experimental/jax2tf/impl_no_xla.py

```diff
@@ -851,10 +855,14 @@ def sparse_scatter(
     wd = dimension_numbers.inserted_window_dims
     sd = dimension_numbers.scatter_dims_to_operand_dims
     ud = dimension_numbers.update_window_dims
-    tf.assert_equal(tf.rank(operand), tf.rank(updates)+1, message='Updates must be 1 dimension less than operand')
-    tf.assert_rank(scatter_indices, 1, message='Scatter indices must be 1 dimensional')
-    assert (len(wd) == len(sd) == 1) & (wd == sd) & (ud == tuple(range(max(ud)+1))), 'Unsupported scatter op'
-    tf.assert_rank(operand, len(ud)+1, message='Unsupported scatter op')
+    dtype = operand.dtype # assume updates has same dtype as operand
+    if dtype in [tf.bool, tf.complex64]:
+      raise error(f"Scatter does not support operands of type {dtype}")
+    if not ((scatter_indices.shape==(1,)) & (len(operand.shape) == len(updates.shape)+1) & (len(updates.shape) == len(ud)) &
+      (len(wd) == len(sd) == 1) & (wd == sd) & (ud == tuple(range(max(ud)+1)))):
+      raise error('Scatters of more than one row in one dimension are unsupported')
```

Comment:
I think as long as the `lax.scatter_*` is being used correctly the first check `(scatter_indices.shape==(1,)` is all that's needed. The other inputs will only be wrong if someone is calling via lax.scatter_* directly rather than the .at[].* or numpy syntax which is rare and discouraged.

Should I make it just the first bit, or keep all these checks which ask if the scatter is valid in general.

Reply:
I would keep all the checks, this makes it more locally-obvious that the code is correct